bot/controllers/dice.js

```diff
@@ -14,78 +14,46 @@ class DiceController extends BaseController {
     this.commands = [
       new Command(
         '!roll',
-        '!roll <die_type>',
+        '!roll <roll_count><die_type>,[roll_count][die_type]',
         'Rolls the dice',
-        'Rolls a die based on the user input',
+        'Rolls dice based on the user input. Must be in #D# format. Example: 1D20 rolls a single D20.',
         this.rollAction.bind(controller),
       ),
-      new Command(
-        '!listDice',
-        '!listDice',
-        'Lists All Dice',
-        'Sends the user a dm containing a list of all the dice available.',
-        this.listDice.bind(controller),
-        'dm',
-      ),
     ];
   }
 
   // Perform the rolling of the dice
   rollAction() {
     const { message } = this;
-    const dice = message.parsed[1].split(',');
-    util.log('Parsed message: ', dice, 4);
-    let results = 'You rolled a ';
-
-    // Decide what equation to run based on the dice type
-    switch (dice.toString()) {
-      case 'D4':
-        results += Math.floor((Math.random() * 4) + 1);
-        break;
-      case 'D6':
-        results += Math.floor((Math.random() * 6) + 1);
-        break;
-      case 'D8':
-        results += Math.floor((Math.random() * 8) + 1);
-        break;
-      case 'D10':
-        results += Math.floor((Math.random() * 10) + 1);
-        break;
-      case 'D12':
-        results += Math.floor((Math.random() * 12) + 1);
-        break;
-      case 'D20':
-        results += Math.floor((Math.random() * 20) + 1);
-        break;
-      default:
-        // Return an error if user didn't input a supported die type
-        results = 'You rolled an unidentified die, please use `!listDice` to see all available dice!';
-        break;
-    }
+    const diceCommand = message.parsed[1].split(',');
+    util.log('Parsed message: ', diceCommand, 4);
+    let total = 0;
+    let rollsString = '';
+
+    diceCommand.map((singleDie) => {
+      // Get the specifics of the roll based on the location of the 'd'
+      let rollCount = singleDie.toLowerCase().substr(0, singleDie.indexOf('d'));
```

Comment:
This line is causing an issue that is crashing the bot when an unexpected input is found.

Because of the way this line is written, it is looking for an indexOf on a lowercase d... however, when I pass an uppercase D (or any other letter that isn't a D for that matter), the indexOf returns an unexpected value which turns the rollCount into a value that is not a number.

Because of this, line 44 (rollCount -= 1;) may not behave as expected and cause the while loop to loop endlessly until the bot runs out of memory.

I think this line should be: 

`let rollCount = singleDie.toLowerCase().substr(0, singleDie.toLowerCase.indexOf('d'));`

However, this will not completely fix the issue. If I put in something other than an uppercase D or lowercase d, the bot will still crash.

You might need to implement some sort of check on the indexOf() to make sure the value is within range. If not, it should return an error indicating the problem with the input.